### DISCUSSION TITLE
IKE Rekey

### DISCUSSION THREAD

**Discussion by hnsantu554** [strongswan/strongswan#2483] [Q&A]
I have configured conn.rekey_time = 8hrs

Will this parameter rekey both ike session key and child session keys ?

Please confirm ,thank you

**Comment by tobiasbrunner** [comment]
Rekeying the IKE_SA has no effect on existing CHILD_SAs, they are adopted by the new IKE_SA.

However, the keys of CHILD_SAs that are negotiated after such a rekeying will be based on the key material of the new IKE_SA. So even if  the CREATE_CHILD_SA exchanges used to rekey or create CHILD_SAs don't use separate key exchanges, the keys of the resulting CHILD_SAs will be derived from fresh key material if they are rekeyed regularly as well.

**Reply by hnsantu554** [reply]
Thanks for the clarification, 
Is there any error notification using error-notify plugin, if the rekey is rejected by other peer?

**Reply by tobiasbrunner** [reply]
Rejected how? And why?

**Reply by hnsantu554** [reply]
I mean like if any problem occurs during rekey ? Will there be any error notification for rekey failed

**Reply by tobiasbrunner** [reply]
What problem are you thinking about? Anyway, if an IKE_SA rekeying fails there really isn't much we can do (we just try again after a short delay, there is no alert for it). For CHILD_SA rekeyings there could be the same alerts that are triggered when creating CHILD_SAs from scratch (e.g. due to proposal or TS mismatches), but there is no specific alert for failed rekeyings (we usually also retry after a delay).
